packages/block-editor/src/components/block-list/block-popover.js

```diff
@@ -121,6 +113,14 @@ function BlockPopover( {
 		initialToolbarItemIndexRef.current = undefined;
 	}, [ clientId ] );
 
+	const [ selectedElement, setSelectedElement ] = useState();
+	const [ lastSelectedElement, setLastSelectedElement ] = useState();
+	const [ capturingElement, setCapturingElement ] = useState();
+
+	useBlockRef( clientId, setSelectedElement );
+	useBlockRef( lastClientId, setLastSelectedElement );
+	useBlockRef( capturingClientId, setCapturingElement );
```

Comment:
In most cases, refs are used without "state" to avoid rerendering when they change. This hook forces the opposite, is this what we we want? rerender this when the ref changes?

Reply:
Yes, previously it did the same. We can only render the popover once we know the block element.